~~i do not uderstand why is `aquasecurity/trivy-action` failing for this rock.. ? it's clearly seeing the trivyignore (see part of logs below) and then just exiting with `1`.~~

EDIT: CVE-2024-49761 (rexml) https://github.com/canonical/ruby-rock/actions/runs/20259868891/attempts/6#summary-58185792153. thanks @zhijie-yang!

for reference, here is the code for that step: https://github.com/canonical/oci-factory/blob/574052901c73aae6611e2915904a3826fa7a5c46/.github/workflows/Test-Rock.yaml#L308C1-L329C65

```
Found ignorefile 'rock-repo/.trivyignore':
# Upstream CVEs

# net-imap: net-imap rubygem vulnerable to possible DoS by memory exhaustion
# https://ubuntu.com/security/CVE-2025-43857
CVE-2025-43857

# WEBrick: HTTP request smuggling
# https://ubuntu.com/security/CVE-2024-47220
CVE-2024-47220

# REXML DoS vulnerability when parsing XML containing multiple XML declarations
# https://ubuntu.com/security/CVE-2025-58767
CVE-2025-58767Running Trivy with options: trivy image test-img:test
2025-12-16T10:36:44Z	INFO	Adding schema version to the DB repository for backward compatibility	repository="public.ecr.aws/aquasecurity/trivy-db:2"
2025-12-16T10:36:44Z	INFO	Adding schema version to the DB repository for backward compatibility	repository="index.docker.io/aquasec/trivy-db:2"
2025-12-16T10:36:44Z	INFO	Adding schema version to the DB repository for backward compatibility	repository="ghcr.io/aquasecurity/trivy-db:2"
2025-12-16T10:36:44Z	INFO	Adding schema version to the DB repository for backward compatibility	repository="public.ecr.aws/aquasecurity/trivy-java-db:1"
2025-12-16T10:36:44Z	INFO	Adding schema version to the DB repository for backward compatibility	repository="index.docker.io/aquasec/trivy-java-db:1"
2025-12-16T10:36:44Z	INFO	Adding schema version to the DB repository for backward compatibility	repository="ghcr.io/aquasecurity/trivy-java-db:1"
2025-12-16T10:36:44Z	INFO	[vuln] Vulnerability scanning is enabled
2025-12-16T10:36:44Z	INFO	[secret] Secret scanning is enabled
2025-12-16T10:36:44Z	INFO	[secret] If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2025-12-16T10:36:44Z	INFO	[secret] Please see also https://aquasecurity.github.io/trivy/v0.56/docs/scanner/secret#recommendation for faster secret detection
2025-12-16T10:36:47Z	WARN	No OS package is detected. Make sure you haven't deleted any files that contain information about the installed packages.
2025-12-16T10:36:47Z	WARN	e.g. files under "/lib/apk/db/", "/var/lib/dpkg/" and "/var/lib/rpm"
2025-12-16T10:36:47Z	INFO	Detected OS	family="ubuntu" version="24.04"
2025-12-16T10:36:47Z	INFO	[ubuntu] Detecting vulnerabilities...	os_version="24.04" pkg_num=0
2025-12-16T10:36:47Z	INFO	Number of language-specific files	num=1
2025-12-16T10:36:47Z	INFO	[gemspec] Detecting vulnerabilities...
2025-12-16T10:36:47Z	WARN	Using severities from other vendors for some vulnerabilities. Read https://aquasecurity.github.io/trivy/v0.56/docs/scanner/vulnerability#severity-selection for details.
Error: Process completed with exit code 1.
```

--- a/.trivyignore
+++ b/.trivyignore
@@ -11,3 +11,9 @@ CVE-2024-47220
 # REXML DoS vulnerability when parsing XML containing multiple XML declarations
 # https://ubuntu.com/security/CVE-2025-58767
 CVE-2025-58767
+
+# The REXML gem before 3.3.9 has a ReDoS vulnerability when it parses an XML 
+# that has many digits between &# and x...; in a hex numeric character 
+# reference (&#x...;).
+# https://ubuntu.com/security/CVE-2024-49761
+CVE-2024-49761

--- a/ruby/3.2-24.04/rockcraft.yaml
+++ b/ruby/3.2-24.04/rockcraft.yaml
@@ -21,7 +21,7 @@ platforms:
 
 environment:
   RUBYOPT: "--disable-did_you_mean --disable-error_highlight --disable-syntax_suggest"
-  GEM_PATH: "/usr/lib/ruby/gems/3.3.0/gems:/usr/share/rubygems-integration/all"
+  GEM_PATH: "/usr/lib/ruby/gems/3.2.0/gems:/usr/share/rubygems-integration/all"
 
 parts:
   ruby:

